src/lib/cookie/cookie.js

```diff
@@ -1,51 +1,72 @@
 import Promise from 'promise-polyfill';
 import log from '../log';
+import { TCString } from '@iabtcf/core';
 import {
 	padRight,
 	encodeVendorCookieValue,
-	decodeVendorCookieValue,
-	encodePublisherCookieValue,
-	decodePublisherCookieValue
 } from './cookieutils';
-
-import { sendPortalCommand } from '../portal';
-import pack from '../../../package.json';
-import config from '../config';
 const arrayFrom = require('core-js/library/fn/array/from');
 
-const PUBLISHER_CONSENT_COOKIE_NAME = 'pubconsent';
-const PUBLISHER_CONSENT_COOKIE_MAX_AGE = 33696000;
+const CONSENT_COOKIE = 'adpconsent';
+const CONSENT_COOKIE_MAX_AGE = 33696000;
+const MAX_PURPOSE_V1_ID = 5;
 
-const VENDOR_CONSENT_COOKIE_NAME = 'euconsent';
-const VENDOR_CONSENT_COOKIE_MAX_AGE = 33696000;
+function readCookie(name) {
+	const value = `; ${document.cookie}`;
+	const parts = value.split(`; ${name}=`);
 
+	if (parts.length === 2) {
+		return parts.pop().split(';').shift();
+	}
+}
 
-function encodeVendorIdsToBits(maxVendorId, selectedVendorIds = new Set()) {
-	let vendorString = '';
-	for (let id = 1; id <= maxVendorId; id++) {
-		vendorString += (selectedVendorIds.has(id) ? '1' : '0');
+function writeCookie(name, value, maxAgeSeconds, path = '/') {
+	const maxAge = maxAgeSeconds === null ? '' : `;max-age=${maxAgeSeconds}`;
+	document.cookie = `${name}=${value};path=${path}${maxAge}`;
+}
+
+function decodeConsentData (encoded) {
+	let decoded;
+	try {
+		decoded = TCString.decode(encoded);
+	} catch (e) {
+		log.debug('Consent string not compatible with TCF v2.0');
 	}
-	return padRight(vendorString, Math.max(0, maxVendorId - vendorString.length));
+	return decoded;
 }
 
-function encodePurposeIdsToBits(purposes, selectedPurposeIds = new Set()) {
+function encodeConsentData (decoded) {
+	return TCString.encode(decoded);
+}
+
+function readConsentCookie () {
+	const cookie = readCookie(CONSENT_COOKIE);
+	log.debug('Read consent data from local cookie', cookie);
+	return Promise.resolve(cookie && decodeConsentData(cookie));
+}
+
+function writeConsentCookie(encodedConsent) {
+	log.debug('Write consent data to local cookie');
+	return Promise.resolve(writeCookie(CONSENT_COOKIE, encodedConsent, CONSENT_COOKIE_MAX_AGE, '/'));
+}
```

Comment:
Hint: TCFv2 doesn't require that consent must be stored in cookie. So I think we may even remove whole this logic from here and force publisher to always pass `getConsentData` and `setConsentData` functions in configuration.